cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/system/SystemWorkerServiceLauncher.java

```diff
@@ -0,0 +1,182 @@
+  }
+
+  public void run() {
+    TwillController activeController = null;
+    for (TwillController controller : twillRunner.lookup(SystemWorkerTwillApplication.NAME)) {
+      // If detected more than one controller, terminate those extra controllers.
+      if (activeController != null) {
+        controller.terminate();
+      } else {
+        activeController = controller;
+      }
+    }
+    // If there is no system worker runner running, create one
+    if (activeController == null) {
+      try {
+        Path tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
+            cConf.get(Constants.AppFabric.TEMP_DIR)).toPath();
+        Files.createDirectories(tmpDir);
+
+        Path runDir = Files.createTempDirectory(tmpDir, "system.worker.launcher");
+        try {
+          // Unset the internal certificate path since certificate is stored cdap-security which
+          // is not exposed (i.e. mounted in k8s) to SystemWorkerService.
+          CConfiguration cConfCopy = CConfiguration.copy(cConf);
+          cConfCopy.unset(Constants.Security.SSL.INTERNAL_CERT_PATH);
+          Path cConfPath = runDir.resolve("cConf.xml");
+          try (Writer writer = Files.newBufferedWriter(cConfPath, StandardCharsets.UTF_8)) {
+            cConfCopy.writeXml(writer);
+          }
+          Path hConfPath = runDir.resolve("hConf.xml");
+          try (Writer writer = Files.newBufferedWriter(hConfPath, StandardCharsets.UTF_8)) {
+            hConf.writeXml(writer);
+          }
+
+          ResourceSpecification systemResourceSpec = ResourceSpecification.Builder.with()
+              .setVirtualCores(cConf.getInt(Constants.SystemWorker.CONTAINER_CORES))
+              .setMemory(cConf.getInt(Constants.SystemWorker.CONTAINER_MEMORY_MB), ResourceSpecification.SizeUnit.MEGA)
+              .setInstances(cConf.getInt(Constants.SystemWorker.CONTAINER_COUNT))
+              .build();
+
+          ResourceSpecification artifactLocalizerResourceSpec = ResourceSpecification.Builder.with()
+              .setVirtualCores(cConf.getInt(Constants.ArtifactLocalizer.CONTAINER_CORES))
+              .setMemory(cConf.getInt(Constants.ArtifactLocalizer.CONTAINER_MEMORY_MB),
+                  ResourceSpecification.SizeUnit.MEGA)
+              .setInstances(cConf.getInt(Constants.TaskWorker.CONTAINER_COUNT))
+              .build();
+
+          LOG.info("Starting SystemWorker pool with {} instances", systemResourceSpec.getInstances());
+
+          TwillPreparer twillPreparer = twillRunner.prepare(
+              new SystemWorkerTwillApplication(cConfPath.toUri(), hConfPath.toUri(), systemResourceSpec,
+                  artifactLocalizerResourceSpec));
+
+          String priorityClass = cConf.get(Constants.TaskWorker.CONTAINER_PRIORITY_CLASS_NAME);
+          if (priorityClass != null) {
+            twillPreparer = twillPreparer.setSchedulerQueue(priorityClass);
+          }
+
+          if (twillPreparer instanceof DependentTwillPreparer) {
```

Comment:
We also need to setup SecureTwillPreparer with withSecretDisk for both containers.

Reply:
As discussed with @masoud-io, this will be done as part of Dispatch pipeline PR.